### PR TITLE
Feat/remove language tool package

### PR DESCRIPTION
**Pull request by silverkate**

### CODE REVIEW

File: lib/client/language_tool_client.dart [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    'Accept': 'application/json',
+  };
+
+  /// Constructor for [LanguageToolClient].
+  LanguageToolClient();
+
+  /// Checks the errors in text.
+  Future<List<WritingMistake>> check(
+    String text, {
+    String language = 'auto',
```
Comment (solid-danylokhvan) [inline lib/client/language_tool_client.dart:26]: Just wanted to confirm if we want to keep the autoselected language feature. I noticed that it sometimes picks up wrong things in English sentences.

Снимок экрана 2023-06-27 в 20 20 52

Comment (solid-vovabeloded) [reply lib/client/language_tool_client.dart:26]: I think we want to add a language selection feature in a separate PR after this one.

File: lib/beans/match.dart
Comment (solid-vovabeloded) [inline lib/beans/match.dart:1]: Let's rename `beans` => `model` and move it to the `core` folder.

File: lib/beans/mistake_context.dart [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/// A data model class that stores context of the error,
+/// i.e. the error and some text to the left and to the left.
+class MistakeContext {
```
Comment (solid-vovabeloded) [inline lib/beans/mistake_context.dart:3]: Do we need this model? Maybe it makese sense to remove it?

File: lib/core/enums/mistake_type.dart [hunk 3]
```diff
@@ -19,5 +19,15 @@ enum MistakeType {
...
   /// Any other mistake type
-  other,
+  other;
+
+  /// Getting the [MistakeType] from String.
+  static MistakeType fromString(String value) {
+    final lowercasedValue = value.toLowerCase();
+
+    return MistakeType.values.firstWhere(
+      (type) => type.toString().split('.').last == lowercasedValue,
```
Comment (solid-vovabeloded) [inline lib/core/enums/mistake_type.dart:29]: Would it be better to have something like a `value` field in this enum and use it to parse the enum value from a string?

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]

**Review by solid-vovabeloded — APPROVED** [review]
Please, resolve this suggestion and you'll be good to go.

**Pull request merged**